x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecyclePolicy.java

```diff
@@ -102,7 +102,7 @@ public SnapshotLifecyclePolicy(StreamInput in) throws IOException {
         this.schedule = in.readString();
         this.repository = in.readString();
         this.configuration = in.readMap();
-        this.retentionPolicy = new SnapshotRetentionConfiguration(in);
+        this.retentionPolicy = in.readOptionalWriteable(SnapshotRetentionConfiguration::new);
```

Comment:
I didn't notice this the first time around but we'll probably need to version-gate this if there's a version of SLM that ships that doesn't have retention.

Reply:
+1 on addressing this, but probably not until we're about to merge